### ISSUE TITLE
Additional axis added for heirarchical variables

### ISSUE THREAD

**Issue by roblem** [pymc-devs/pymc4#222]
I noticed this when some custom tensorflow ops inside a model statement stopped working when I added a hierarchical variable.  Please pardon the nonsensical models with no likelihood or observed data, but I believe it is sufficient for illustrating the issue.

- First model, no hierarchical variables and printing out shape and sampling once:
```python
@pm4.model()
def toy_model1():
    b = yield pm4.Normal('b', 0, 10, plate=1)
    sd = yield pm4.Exponential("epsilon", rate=.1, plate=1)
    print("Shape of b:", b.shape)
    print("Shape of sd: ", sd.shape)
``` 
gives expected shapes
```
Shape of b: (1,)
Shape of sd:  (1,)
```
- Second model has a hierarchical variable
```python
@pm4.model()
def toy_model2():
    b = yield pm4.Normal('b', 0, 10, plate=1)
    sd = yield pm4.Exponential("epsilon", rate=.1, plate=1)
    print("Shape of b:", b.shape)
    print("Shape of sd: ", sd.shape)
    b_ = yield pm4.Normal('b_', b, sd, plate=10)
    print("Shape of b_:", b_.shape)
```
gives unexpected shape (to me at least) for `b_`:
```
Shape of b: (1,)
Shape of sd:  (1,)
Shape of b_: (1, 10)
```
I would have expected the shape to be (10,).

**Comment by lucianopaz** [comment]
Thanks for reporting this. I mentioned this problem when I opened up #193, but didn't open a corresponding issue. Regardless, once we finish #193, the issue you are seeing will be fixed.

The only thing to note is that we will change the `plate` argument to `batch_stack`

**Comment by lucianopaz** [comment]
Closed by #193
